flox-file-parser/app/lib/parser.movies.js

```diff
@@ -0,0 +1,108 @@
+const fs = require("fs")
+const path = require("path")
+const db = require("../../database/models")
+
+const videoNameParser = require("video-name-parser")
+
+const supportedVideoFileTypes = ["mkv", "mp4"]
+const env = process.env
+const { file_history } = db.sequelize.models
+
+const fetchSubtitles = (episodesPath, fileName) => {
+  const subtitlePath = episodesPath + "/" + fileName + ".srt" 
+
+  if (fs.existsSync(subtitlePath)) {
+    return fs.realpathSync(subtitlePath)  
+  }
+
+  return null
+}
+
+const searchDirectory = (path) => {
+  const files = fs.readdirSync(path)
+  const foundFiles = []
+
+  files.forEach((file) => {
+    const currentFile = path + "/" + file
+    const fileInfo = fs.statSync(currentFile)
+
+    if (fileInfo.isDirectory()) {
+      return foundFiles.push(...searchDirectory(currentFile))
+    }
+
+    return foundFiles.push(currentFile)
+  })
+
+  return foundFiles
+}
+
+const addMovie = (file, promises) => {
+  const pathInfo = path.parse(file)
+  const ext = pathInfo.ext.replace(".", "")
+
+  if(!supportedVideoFileTypes.includes(ext)) return
+
+  const fileInfo = videoNameParser(pathInfo.name) 
+  const filePath = pathInfo.dir + "/" + pathInfo.base
+
+  const src = fs.realpathSync(filePath)
+  promises.push(file_history.findOrCreate({
+    where: { 
+      src: src,
+      $and: {
+        removed: null
+      }
+    },
+    defaults: {
+      extension: ext,
+      filename: pathInfo.name,
+      name: fileInfo.name,
+      src: src,
+      year: fileInfo.year,
+      tags: fileInfo.tag.toString(),
+      subtitles: fetchSubtitles(pathInfo.dir, pathInfo.name),
+      category: "movies"
+    }
+  }))
+}
+
+const removeMovies = (list, dbPromises) => {
+  list.forEach((missingMovie) => {
+    dbPromises.push(file_history.update({ removed: Date.now() }, {
+      where: { src: missingMovie, $and: { removed: null } }
+    }))
+  })
+}
+
+const updateMovies = (ParserList) => {
+  const { MOVIES_ROOT } = env
+  const listOfMoviesInDB = ParserList().filter((file) => file.category === "movies" && file.removed == null)
+  const allFiles = searchDirectory(MOVIES_ROOT)
+
+  const dbPromises = []
+
+  return listOfMoviesInDB.then((l) => {
+    const list = l.map((m) => m.src)
```

Comment:
In `parse.tv.js` this `map` is used directly on `listOfMoviesInDB/listOfTvInDB`. Is there a reason for not doing it here also?